src/data/pokeball.ts

```diff
@@ -82,11 +83,38 @@ export function getPokeballTintColor(type: PokeballType): number {
   }
 }
 
-export function doPokeballBounceAnim(scene: BattleScene, pokeball: Phaser.GameObjects.Sprite, y1: number, y2: number, baseBounceDuration: integer, callback: Function) {
+/**
+ * Gets the critical capture chance based on number of mons registered in Dex and modified {@link https://bulbapedia.bulbagarden.net/wiki/Catch_rate Catch rate}
+ * Formula from {@link https://www.dragonflycave.com/mechanics/gen-vi-vii-capturing Dragonfly Cave Gen 6 Capture Mechanics page}
+ * @param scene {@linkcode BattleScene} current BattleScene
+ * @param modifiedCatchRate the modified catch rate as calculated in {@linkcode AttemptCapturePhase}
+ * @returns the chance of getting a critical capture, out of 256
+ */
+export function getCriticalCaptureChance(scene: BattleScene, modifiedCatchRate: number): number {
+  if (scene.gameMode.isDaily || scene.gameMode.isFreshStartChallenge()) {
+    return 0;
+  }
+  const dexCount = scene.gameData.getSpeciesCount(d => !!d.caughtAttr);
+  const catchingCharmMultiplier = new NumberHolder(1);
+  //scene.findModifier(m => m instanceof CriticalCatchChanceBoosterModifier)?.apply(catchingCharmMultiplier);
+  const dexMultiplier = dexCount > 800 ? 2.5
+    : dexCount > 600 ? 2
+      : dexCount > 400 ? 1.5
+        : dexCount > 200 ? 1
+          : dexCount > 100 ? 0.5
+            : 0;
+  return Math.floor(catchingCharmMultiplier.value * dexMultiplier * Math.min(255, modifiedCatchRate) / 6);
+}
+
+export function doPokeballBounceAnim(scene: BattleScene, pokeball: Phaser.GameObjects.Sprite, y1: number, y2: number, baseBounceDuration: number, callback: Function, isCritical: boolean = false) {
   let bouncePower = 1;
   let bounceYOffset = y1;
   let bounceY = y2;
   const yd = y2 - y1;
+  const x0 = pokeball.x;
+  const x1 = x0 + 4;
+  const x2 = x0 - 4;
```

Comment:
```suggestion
  const x1 = x0 + 3;
  const x2 = x0 - 3;
```
Makes the critical capture shake less intense